### PR TITLE
Finish updating docstrings

### PR DESCRIPTION
**Pull request by brynpickering**
Updated docstrings I missed in #229 and added type hints to methods if they have an associated docstring. Now everything will render nicely in the docs :)

### CODE REVIEW

**Review by KasiaKoz — CHANGES_REQUESTED** [review]
Looks good, there are just a couple of small things:
- I think the description of the spatial tree is a bit off right now
- multiprocessing description needs a little ironing in one place, it mentioned `combine` as somewhat optional which might give the wrong idea (i think you still need to pass a method, even if that method is an identity map)

I think I left some other comments but those will be mostly just passing thoughts

File: genet/utils/spatial.py [hunk 1]
```diff
@@ -13,82 +13,112 @@
...
 
 
-def encode_shapely_linestring_to_polyline(linestring):
+def encode_shapely_linestring_to_polyline(linestring: LineString) -> str:
     """
-    :param linestring: shapely.geometry.LineString
-    :return: google encoded polyline
+
+    Args:
+        linestring (LineString): _description_
```
[the remark this answers is not in the document]
Comment (brynpickering) [reply genet/utils/spatial.py:41]: They are included when autogenerating docs using [this VSCode extension](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring)

**Comment by brynpickering** [comment]
@KasiaKoz bump to avoid this going stale!

**Review by KasiaKoz — APPROVED** [review]
👍

**Pull request merged**